src/compile.ts

```diff
@@ -525,6 +531,96 @@ export function compile(
         ]);
       }
 
+      function visitApiIntegrationsStaticMethod(
+        node: ApiIntegrationsStaticMethodInterface
+      ): ts.CallExpression {
+        const [props] = node.arguments;
+
+        return ts.factory.updateCallExpression(
+          node,
+          node.expression,
+          node.typeArguments,
+          [visitApiIntegrationProps(props)]
+        );
+      }
+
+      function visitApiIntegration(node: ApiIntegrationInterface): ts.Node {
+        const [props] = node.arguments;
+
+        return ts.factory.updateNewExpression(
+          node,
+          node.expression,
+          node.typeArguments,
+          [visitApiIntegrationProps(props)]
+        );
+      }
+
+      function visitApiIntegrationProps(
+        props: ts.ObjectLiteralExpression
+      ): ts.ObjectLiteralExpression {
+        return ts.factory.updateObjectLiteralExpression(
+          props,
+          props.properties.map((prop) => {
+            if (
+              ts.isPropertyAssignment(prop) &&
+              (ts.isStringLiteral(prop.name) || ts.isIdentifier(prop.name))
+            ) {
+              if (
+                prop.name.text === "responses" ||
+                prop.name.text === "errors"
+              ) {
+                return visitApiIntegrationResponsesProp(prop);
+              } else if (
+                prop.name.text === "request" ||
+                prop.name.text === "response"
+              ) {
+                return visitApiIntegrationMapperProp(prop);
+              }
+            }
+            return prop;
+          })
+        );
+      }
+
+      function visitApiIntegrationResponsesProp(
+        prop: ts.PropertyAssignment
+      ): ts.PropertyAssignment {
+        const { initializer } = prop;
+        if (!ts.isObjectLiteralExpression(initializer)) {
+          return prop;
+        }
+
+        return ts.factory.updatePropertyAssignment(
+          prop,
+          prop.name,
+          ts.factory.updateObjectLiteralExpression(
+            initializer,
+            initializer.properties.map((p) =>
+              ts.isPropertyAssignment(p) ? visitApiIntegrationMapperProp(p) : p
+            )
+          )
+        );
+      }
+
+      function visitApiIntegrationMapperProp(
+        prop: ts.PropertyAssignment
+      ): ts.PropertyAssignment {
+        const { initializer } = prop;
+        const toFunc = errorBoundary(() => {
+          if (
+            !ts.isFunctionExpression(initializer) &&
+            !ts.isArrowFunction(initializer)
+          ) {
+            throw new Error(
+              `Expected mapping property of an ApiIntegration to be a function. Found ${initializer.getText()}.`
+            );
+          }
+          return toFunction("FunctionDecl", initializer);
```

Comment:
Hmm we definitely need a more general way of identifying what should be transformed. A ton of plumbing just to get here.